### PR TITLE
Adds documentation for JENKINS_UC_INSECURE variable

### PR DESCRIPTION
**Pull request by akram**
Adds documentation for JENKINS_UC_INSECURE variable

### CODE REVIEW

**Comment by akram** [comment]
@gabemontero and @bmcelvee could you please take a look?

this is related to:
https://github.com/openshift/jenkins/pull/866

**Comment by akram** [comment]
Another PR from @bmcelvee is pending for the 4.x branch which adds Jenkins environment variable. So I created a PR on top of @bmcelvee https://github.com/bmcelvee/openshift-docs/pull/2

**Comment by akram** [comment]
@wewang58 @xiuwang can you PTAL ?

File: using_images/other_images/jenkins.adoc [hunk 1]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
...
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy  
+performs man-in-the-middle interceptions. This variable applies to plugins downloads which may 
+occur during Jenkins image build or if an extension of the Jenkins image is built or if you 
+run the jenkins image and leverage one of the options to download additional plugins 
+(use of s2i whith plugins.txt or use of `INSTALL_PLUGINS` environment variable.
+To enable, set to `true`.
```
Comment (xiuwang) [inline using_images/other_images/jenkins.adoc]: missing ‘)’

File: using_images/other_images/jenkins.adoc [hunk 2]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
 the CentOS or RHEL version of the Jenkins image. This variable must be set before Jenkins starts
 the first time for it to have an effect.
 
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy  
+performs man-in-the-middle interceptions. This variable applies to plugins downloads which may 
+occur during Jenkins image build or if an extension of the Jenkins image is built or if you
```
Comment (xiuwang) [inline using_images/other_images/jenkins.adoc]: If the upper case “Jenkins” here are necessary?

**Comment by wewang58** [comment]
others lgtm

File: using_images/other_images/jenkins.adoc (hunk 2 above)
Comment (akram) [reply using_images/other_images/jenkins.adoc]: fixed, I set Jenkins with upper everywhere.

File: using_images/other_images/jenkins.adoc (hunk 1 above)
Comment (akram) [reply using_images/other_images/jenkins.adoc]: Fixed

Comment (xiuwang) [reply using_images/other_images/jenkins.adoc]: @akram After setting INSTALL_PLUGINS with specified version plugins, the jenkins s2i build didn't install the versioned plugins.
Could you make the INSTALL_PLUGINS  usage more clear? Thanks~
$ oc build-logs  jenkins-master-2  | grep "openshift-sync"
Command "build-logs" is deprecated, use oc logs build/<build-name>
STEP 3: ENV OPENSHIFT_BUILD_NAME="jenkins-master-2" OPENSHIFT_BUILD_NAMESPACE="xiu" OPENSHIFT_BUILD_SOURCE="https://github.com/xiuwang/jenkins-example.git" OPENSHIFT_BUILD_REFERENCE="Jenkins_uc_insecure" OPENSHIFT_BUILD_COMMIT="e6941401a68da0016e761fe98fdf1681132e551f" JENKINS_UC="https://***" JENKINS_UC_INSECURE="true" INSTALL_PLUGINS="openshift-sync:1.0.37"
openshift-sync:1.0.38

**Comment by xiuwang** [comment]
The feature works, and doc LGTM , Thanks

**Comment by vikram-redhat** [comment]
@bmcelvee - can this be merged?

**Comment by bmcelvee** [comment]
@vikram-redhat yes, thanks!

File: using_images/other_images/jenkins.adoc [hunk 3]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
 the CentOS or RHEL version of the Jenkins image. This variable must be set before Jenkins starts
 the first time for it to have an effect.
 
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/unknow/unknown

File: using_images/other_images/jenkins.adoc [hunk 4]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
...
 
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy  
+performs man-in-the-middle interceptions. This variable applies to plugins downloads which may 
+occur during Jenkins image build or if an extension of the Jenkins image is built or if you 
+run the Jenkins image and leverage one of the options to download additional plugins 
+(use of s2i whith plugins.txt or use of `INSTALL_PLUGINS` environment variable).
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/whith plugins.txt/with `plugins .txt`

File: using_images/other_images/jenkins.adoc [hunk 5]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
...
 the first time for it to have an effect.
 
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy  
+performs man-in-the-middle interceptions. This variable applies to plugins downloads which may 
+occur during Jenkins image build or if an extension of the Jenkins image is built or if you 
+run the Jenkins image and leverage one of the options to download additional plugins
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: s/plugins/plug-ins

File: using_images/other_images/jenkins.adoc [hunk 6]
```diff
@@ -293,6 +293,16 @@ The default maven agent pod uses `docker.io/openshift/jenkins-agent-maven-35-cen
 the CentOS or RHEL version of the Jenkins image. This variable must be set before Jenkins starts
 the first time for it to have an effect.
 
+* `JENKINS_UC_INSECURE`
++
+Determines whether Jenkins plugins downloads are allowed if the Jenkins Update Center repository 
+uses an invalid SSL certificate. This could be the case if a self hosted repository 
+using self-signed certificate with an unknow CA is used or if an enteprise proxy  
+performs man-in-the-middle interceptions. This variable applies to plugins downloads which may
```
Comment (bmcelvee) [inline using_images/other_images/jenkins.adoc]: I would suggest changing this sentence: "This variable applies to plugins downloads which may occur during Jenkins image build or if an extension of the Jenkins image is built or if you run the Jenkins image and leverage one of the options to download additional plugins (use of s2i whith plugins.txt or use of INSTALL_PLUGINS environment variable). To enable, set to true." 

To: 

"This variable applies to plug-in downloads, which may occur during a Jenkins image build or if an extension of the Jenkins image is built. It is also applied when you run the Jenkins image and use one of the options to download additional plug-ins, including S2I with `plugins.txt` or the `INSTALL_PLUGINS` environment variable. Set to true to enable this variable."

**Comment by bmcelvee** [comment]
@akram Thank you again for submitting this PR! I have two suggested changes: a typo correction and sentence re-write. If you wouldn't mind making those changes, then squashing your commits. I'll go ahead and get this PR merged and cherrypicked appropriately. 

Thanks again!

**Comment by akram** [comment]
Hi @bmcelvee 
just pushed the typo and rephrase change.

**Comment by bmcelvee** [comment]
@akram - thanks! I'll go ahead and merge and cherrypick to 3.11. How many versions back are the changes applicable?

**Comment by akram** [comment]
> @akram - thanks! I'll go ahead and merge and cherrypick to 3.11. How many versions back are the changes applicable?

only for 3.11

**Comment by bmcelvee** [comment]
Perfect! We should be all set on this one, then. Thanks again!

**Pull request merged**